wizard/src/components/mapping_column_options.tsx

```diff
@@ -0,0 +1,211 @@
+                    <option value={type} key={type}>
+                      {props.placeDetector.placeTypes.get(type).displayName}
+                    </option>
+                  ))}
+                </Input>
+                and of format
+                <Input
+                  id="column-type-place-property"
+                  type="select"
+                  value={props.column.columnPlaceProperty.dcid}
+                  onChange={(e) =>
+                    updateColumn(
+                      MappingType.COLUMN,
+                      MappedThing.PLACE,
+                      undefined,
+                      e.target.value
+                    )
+                  }
+                >
+                  {validPlaceProperties.map((property) => (
+                    <option value={property} key={property}>
+                      {
+                        props.placeDetector.placeProperties.get(property)
+                          .displayName
+                      }
+                    </option>
+                  ))}
+                </Input>
+              </>
+            )}
+          </div>
+        </Label>
+        <Label radio="true" className="column-options-input header-type">
+          <Input
+            type="radio"
+            name="mapping-type"
+            checked={props.column.type === MappingType.COLUMN_HEADER}
+            onChange={() =>
+              updateColumn(
+                MappingType.COLUMN_HEADER,
+                props.column.headerMappedThing
+              )
+            }
+          />
+          {
+            <>
+              <span>Data values for the </span>
+              <Input
+                id="header-type-mapped-thing"
+                type="select"
+                value={props.column.headerMappedThing}
+                onChange={(e) =>
+                  updateColumn(
+                    MappingType.COLUMN_HEADER,
+                    e.target.value as MappedThing
+                  )
+                }
+                className="pac-target-input"
+              >
+                {Object.values(MappedThing).map((thing) => (
+                  <option value={thing} key={thing}>
+                    {MAPPED_THING_NAMES[thing] || thing}
+                  </option>
+                ))}
+              </Input>
+              <span>{props.column.column.header}</span>
+            </>
+          }
+        </Label>
+        <Label radio="true" className="column-options-input no-type">
+          <Input
+            type="radio"
+            name="mapping-type"
+            checked={_.isEmpty(props.column.type)}
+            onChange={() => updateColumn(null, null)}
+          />
+          Not mapped
+        </Label>
+      </FormGroup>
+    </div>
+  );
+
+  function updateColumn(
+    mappingType: MappingType,
+    mappedThing: MappedThing,
+    columnPlaceType?: string,
+    columnPlaceProperty?: string
+  ): void {
+    const updatedColumn = _.cloneDeep(props.column);
```

Comment:
Where is props being used from? It's not provided as a function arg and I don't see a global variable?

Reply:
I see